File: IntroduccionPoo/Ejercicio3/Coche.java
package IntroduccionPoo.Ejercicio3;

class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }

    // a) Acelerar
    public void acelerar() {
        velocidad += 10;
    }

    // b) Frenar
    public void frenar() {
        velocidad -= 5;
    }

    public void mostrarVelocidad() {
        System.out.println(marca + " " + modelo + " va a " + velocidad + " km/h");
    }

    public static void main(String[] args) {
        Coche c1 = new Coche("Toyota", "Corolla");
        Coche c2 = new Coche("Ford", "Mustang");

        c1.acelerar();
        c1.acelerar();
        c1.frenar();
        c1.mostrarVelocidad();

        c2.acelerar();
        c2.mostrarVelocidad();
    }
}

File: IntroduccionPoo/Ejercicio3/Coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    # a) Acelerar
    def acelerar(self):
        self.velocidad += 10

    # b) Frenar
    def frenar(self):
        self.velocidad -= 5

    def mostrar_velocidad(self):
        print(f"{self.marca} {self.modelo} va a {self.velocidad} km/h")


if __name__ == "__main__":
    c1 = Coche("Toyota", "Corolla")
    c2 = Coche("Ford", "Mustang")

    c1.acelerar()
    c1.acelerar()
    c1.frenar()
    c1.mostrar_velocidad()

    c2.acelerar()
    c2.mostrar_velocidad()
